src/l500/l500-depth.cpp

```diff
@@ -374,6 +392,67 @@ namespace librealsense
 
     void l500_depth_sensor::start(frame_callback_ptr callback)
     {
+        if (supports_option(RS2_OPTION_HOST_PERFORMANCE))
```

Comment:
What about encapsulating the below code into opt->set(..) ?
The TRB settings are customized per sensor and can be passed into the option at ctor level (where sensor->register_option( occrs), so this could can become:
if supports(TRB)
  trb_opt-set(TRB_MOD)

Reply:
thanks for the suggestion, looked into it, will implement tomorrow. if need to make the build, please go ahead and take what current implementation. can do another PR for this part of the change once it's tested.

Reply:
Thanks, we'll merge the current implementation so that refactoring will be moved to the follow-on task.
@gwen2018 , please open a new PR for the remaining work.